cpg_pipes/targets.py

```diff
@@ -0,0 +1,640 @@
+            sequencing_type=sequencing_type,
+            participant_id=participant_id,
+            meta=meta,
+            sex=sex,
+            pedigree=pedigree,
+            alignment_input=alignment_input,
+        )
+        self._sample_by_id[id] = s
+        return s
+
+    def get_samples(self, only_active: bool = True) -> list['Sample']:
+        """
+        Get dataset's samples. Include only "active" samples, unless only_active=False
+        """
+        return [
+            s for sid, s in self._sample_by_id.items() if (s.active or not only_active)
+        ]
+
+    def get_job_attrs(self) -> dict[str, str]:
+        """
+        Attributes for Hail Batch job.
+        """
+        return {'dataset': self.name}
+
+    def get_job_prefix(self) -> str:
+        """
+        Prefix job names.
+        """
+        return f'{self.name}: '
+
+    def make_ped_file(self) -> Path:
+        """
+        Create a PED file for all samples
+        """
+        datas = []
+        for sample in self.get_samples():
+            if sample.pedigree:
+                datas.append(sample.pedigree.get_ped_dict())
+        df = pd.DataFrame(datas)
+
+        ped_path = self.get_tmp_bucket() / f'{self.name}.ped'
+        with ped_path.open('w') as fp:
+            df.to_csv(fp, sep='\t', index=False)
+
+        return ped_path
+
+
+class Sex(Enum):
+    """
+    Sex as in PED format
+    """
+
+    UNKNOWN = 0
+    MALE = 1
+    FEMALE = 2
+
+    @staticmethod
+    def parse(sex: str | None) -> 'Sex':
+        """
+        Parse a string into a Sex object.
+        """
+        if sex:
+            if sex.lower() in ('m', 'male', '1'):
+                return Sex.MALE
+            if sex.lower() in ('f', 'female', '2'):
+                return Sex.FEMALE
+        return Sex.UNKNOWN
+
+
+class Sample(Target):
+    """
+    Represents a Sample.
+    """
+
+    def __init__(
+        self,
+        id: str,  # pylint: disable=redefined-builtin
+        dataset: 'Dataset',  # type: ignore  # noqa: F821
+        external_id: str | None = None,
+        sequencing_type: SequencingType | None = None,
+        participant_id: str | None = None,
+        meta: dict | None = None,
+        sex: Sex | None = None,
+        pedigree: Optional['PedigreeInfo'] = None,
+        alignment_input: AlignmentInput | None = None,
+    ):
+        super().__init__()
+        self.id = id
+        self._external_id = external_id
+        self.dataset = dataset
+        self.sequencing_type = sequencing_type or SequencingType.GENOME
+        self._participant_id = participant_id
+        self.meta: dict = meta or dict()
+        self.pedigree: PedigreeInfo | None = pedigree
+        if sex:
+            self.pedigree = PedigreeInfo(
+                sample=self,
+                fam_id=self.participant_id,
+                sex=sex,
+            )
+        self.alignment_input: AlignmentInput | None = alignment_input
+
+    def __repr__(self):
+        return (
```

Comment:
Probably not a big deal, but could construct this a bit cleaner. This suggestion probably isn't formatted properly.

```suggestion
    values = {
        'participant': self._participant_id if self._participant_id else '',
        'forced': str(self.forced),
        'active': str(self.active),
        'meta': str(self.meta),
        'sequencing_type': str(self.sequencing_type.value),
        'alignment_input': self.alignment_input if self.alignment_input else ''
        'pedigree': self.pedigree if self.pedigree else ''
    }

    retval = f'Sample({self.dataset.name}/{self.id}'
    if self._external_id:
        retval += f'|{self._external_id}'
    return retval + ''.join(f', {k}={v}' for k, v in values.items())
```

Reply:
That looks much better!